src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -13,95 +13,46 @@ namespace Microsoft.Extensions.Caching.Memory
 {
     internal class CacheEntry : ICacheEntry
     {
-        private bool _disposed;
         private static readonly Action<object> ExpirationCallback = ExpirationTokensExpired;
-        private readonly Action<CacheEntry> _notifyCacheOfExpiration;
-        private readonly Action<CacheEntry> _notifyCacheEntryCommit;
+
+        private readonly object _lock = new object();
+        private readonly MemoryCache _cache;
+
         private IList<IDisposable> _expirationTokenRegistrations;
         private IList<PostEvictionCallbackRegistration> _postEvictionCallbacks;
-        private bool _isExpired;
-        private readonly ILogger _logger;
-
-        internal IList<IChangeToken> _expirationTokens;
-        internal DateTimeOffset? _absoluteExpiration;
-        internal TimeSpan? _absoluteExpirationRelativeToNow;
+        private IList<IChangeToken> _expirationTokens;
         private TimeSpan? _slidingExpiration;
         private long? _size;
         private IDisposable _scope;
         private object _value;
-        private bool _valueHasBeenSet;
+        private int _state; // actually a [Flag] enum called "State"
```

Comment:
Why not actually use `private State _state`?

Reply:
to set the flag in a thread-safe way I need to use `Interlocked` APIs which accept only `int` and `long`